var/spack/repos/builtin/packages/py-numpy/package.py

```diff
@@ -289,12 +320,28 @@ def write_library_dirs(f, dirs):
                         )
                     )
 
+    def setup_dependent_build_environment(self, env, depspec):
+        self.setup_build_environment(env)
+
     def setup_build_environment(self, env):
         # Tell numpy which BLAS/LAPACK libraries we want to use.
         # https://github.com/numpy/numpy/pull/13132
         # https://numpy.org/devdocs/user/building.html#accelerated-blas-lapack-libraries
         spec = self.spec
 
+        if spec['blas'].name != 'intel-mkl' and \
+                spec['blas'].name != 'intel-parallel-studio':
+            env.set('MKLROOT', 'None')
+        if spec['blas'].name != 'blis':
+            env.set('BLIS', 'None')
+        if spec['blas'].name != 'openblas':
+            env.set('OPENBLAS', 'None')
+        if spec['blas'].name != 'atlas':
+            env.set('ATLAS', 'None')
+            env.set('PTATLAS', 'None')
+        if spec['blas'].name != 'veclibfort':
+            env.set('ACCELERATE', 'None')
+
         # https://numpy.org/devdocs/user/building.html#blas
         if 'blas' not in spec:
```

Comment:
I wonder if we could somehow combine these two if-statement blocks, like:
```python
if spec['blas'].name == 'blis':
    blas = 'blis'
else:
    env.set('BLIS', 'None')
if spec['blas'].name == 'openblas':
    blas = 'openblas'
else:
    env.set('OPENBLAS', 'None')
...
```
That would have a bit less duplication.

Reply:
ok